apps/src/p5lab/P5Lab.js

```diff
@@ -238,8 +238,34 @@ P5Lab.prototype.init = function(config) {
   this.level.helperLibraries = this.level.helperLibraries || [];
 
   this.level.softButtons = this.level.softButtons || [];
+  this.startAnimations = {orderedKeys: [], propsByKey: {}};
```

Comment:
This was mostly just me being overly cautious - since start animations gets set in the promise of the `getDefaultListMetadata`, but the rest of this function isn't in a promise structure, I wanted to set a default value for this.startAnimations, so that after this point in the code, whether or not `getDefaultListMetadata` has finished the `then`, we can logic about startAnimations being an object with keys orderedKeys and propsByKey. It's not necessary, but I like the surety it gives. Is it a bad practice?

Reply:
my only worry is that setting a default value may create a bug elsewhere, but i don't know this feature well enough to point to anything specific that could break. if `this.startAnimations` is null/undefined, we know animations haven't loaded yet. but if we set it to an empty value, it looks like there are no animations.

Reply:
Yeah, I think my concern would be if we end up looking for `this.startAnimations.orderedKeys` when this.startAnimations is set to null. But I'm not sure which would be the worse case scenario that we'd want to protect against. :shrug:

Reply:
i think we should handle the case where `this.startAnimations` is undefined since it's loaded asynchronously. it makes it clearer that the animations may or may not be loaded. if `this.startAnimations` is present but empty, i would assume that means there are no animations, not that they haven't loaded yet